OpenTap.Plugins.PNAX/Traces/TraceStatistics.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Author: MyName
+// Copyright:   Copyright 2024 Keysight Technologies
+//              You have a royalty-free right to use, modify, reproduce and distribute
+//              the sample application files (and/or any modified version) in any way
+//              you find useful, provided that you agree that Keysight Technologies has no
+//              warranty, obligations or liability for any sample application files.
+using OpenTap;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+
+namespace OpenTap.Plugins.PNAX
+{
+    [AllowAsChildIn(typeof(TestPlan))]
+    [AllowAsChildIn(typeof(SingleTraceBaseStep))]
+    [Display("Statistics", Groups: new[] { "Network Analyzer", "Trace" }, Description: "Set Statistics for a trace")]
+    public class TraceStatistics : PNABaseStep
+    {
+        #region Settings
+        // Override for Channel so we can call ShowTraceSettings
+        private int _Channel;
+        [EnabledIf("IsControlledByParent", false, HideIfDisabled = false)]
+        [Display("Channel", Order: 0.11)]
+        public override int Channel
+        {
+            get
+            {
+                ShowTraceSettings();
```

Comment:
We did not need call this function in getters. Only in setters when values being updated

Reply:
I want to update this property, I can show you the behavior with and without the function being called in getter. The idea is that TraceStatistics can be used as a child on a SingleTrace and also as a standalone test step in TestPlan root level. If the test is a child under SingleTrace it should use the mnum and channel from the parent step, if the test step is at the root Test Plan level then it should present the user with settings to provide Channel and Mnum.

Reply:
But thanks, this reminded me to update the mnum from parent in case the step is a child under single trace. For root TestPlan level it works fine.

Reply:
Then I think we need to overwrite the IsControlledByParents setter. When it is a child step, it should use mnum. If it is false, then we use settings to provide channel

Reply:
That is exactly what this step is doing. 

I could not detect if the step is a child or at root Test Plan level during the constructor, it always returns null when I query this.Parent. The only way I could find was to use the Getter which is when the step already knows if it is a child or not.

Should I merge this PR and open another issue to review what you are pointing out?

Reply:
Sure. We might need to figure out a better solution here later.